modules/gapi/include/opencv2/gapi/streaming/source.hpp

```diff
@@ -46,14 +45,34 @@ class IStreamSource: public std::enable_shared_from_this<IStreamSource>
     virtual ~IStreamSource() = default;
 };
 
+
+
+namespace detail
+{
 template<class T, class... Args>
-IStreamSource::Ptr inline make_src(Args&&... args)
+typename std::enable_if<!std::is_base_of<IStreamSource, T>::value, IStreamSource::Ptr>::type
+inline make_src_impl(Args&&... args)
+{
+    T builder(std::forward<Args>(args)...);
+    return builder.build();
+}
+
+template<class T, class... Args>
+typename std::enable_if<std::is_base_of<IStreamSource, T>::value, IStreamSource::Ptr>::type
+inline make_src_impl(Args&&... args)
 {
-    static_assert(std::is_base_of<IStreamSource, T>::value,
```

Comment:
I don't get the idea here. You inherit vpl source from the `IStreamSource` but why the assert is removed and how the `enable_if` above (with `builder.build()`) is supposed to be working?